Kernel/Devices/HID/PS2/KeyboardDevice.cpp

```diff
@@ -47,7 +47,29 @@ UNMAP_AFTER_INIT ErrorOr<NonnullOwnPtr<PS2KeyboardDevice>> PS2KeyboardDevice::tr
 
 UNMAP_AFTER_INIT ErrorOr<void> PS2KeyboardDevice::initialize()
 {
-    return attached_controller().reset_device(attached_port_index());
+    ErrorOr<void> err = attached_controller().reset_device(attached_port_index());
+
+    if (err.is_error()) {
+        TRY(attached_controller().send_command(attached_port_index(), SerialIOController::DeviceCommand::GetDeviceID));
+        ErrorOr<u8> res = attached_controller().read_from_device(attached_port_index());
+        if (res.is_error()) {
+            return err;
+        }
+        switch (res.value()) {
+            // Regular and NCD Sun keyboards.
+        case 0xab:
```

Comment:
Hardcoding this stuff here is quite sad, but this will be handled in #18792 - so I put this note here as a remainder to ensure this will be fixed then.
Please don't resolve this comment, we can merge this without fixing this issue for now.